### PR TITLE
Feature/inference i52

### LINKED ISSUE #38: Scale all data to have mean 0 and variance 1 to improve numerical stability for high data dimensionality
**Is your feature request related to a problem? Please describe.**
Right now, EPI doesn't work properly for the stock example if the distributions are too wide. This seems to stem from an numerical issue, where the variances of the data dimensions differ by multiple orders of magnitude, leading to rounding errors stacking up to wrong results. The large support of the pdfs in the data space also leads to extremely small densities, adding to the numerical problems.

**Describe the solution you'd like**
A possible fix, as suggested by @VivianeKlingel for this problem would be to scale all data dimensions to have mean zero and variance one. This transformation wouldn't affect much of the computational base, as it only adds a constant factor to the parameter distribution (see below). However, some additional discussion between the main contributors is required where exactly the transformation should be applied (@TheVincentWagner , @kaiserls ). 

image

**Describe alternatives you've considered**
The alternative of just scaling the data distribution by a constant factor does not seem promising as it doesn't address the issue of large differences of variance in the data space. 

**Additional context**
Estimation results often look like this:

image

### LINKED ISSUE #52: Accelerate data normalization
**Is your feature request related to a problem? Please describe.**
Right now, data normalization can cause EPI to run quite a lot slower (especially for high dimensional problems).

**Describe the solution you'd like**
This might be fixed by using jax to accelerate the corresponding functions.

### LINKED ISSUE #54: Use principal component analysis (PCA) to reduce number of data dimesions
**Is your feature request related to a problem? Please describe.**
High data dimensions cause issues with the distance based kernel-density estimation (see, for instance, https://homes.cs.washington.edu/%7Epedrod/papers/cacm12.pdf). This is likely one of the causes for the problem described in #38 .

**Describe the solution you'd like**
 If the number of parameter dimensions is lower, EPI could use PCA to reduce the number data dimensions (for instance s.t. the number of data dimensions is the number of parameter dimensions). This should retain most of the information contained in the data (especially, the assumption that performing a PCA usually should not break injectivity of the model should still hold).

**Describe alternatives you've considered**
The approach of just normalizing the data to reduce numerical errors as described in #38 did not fully solve the issue.

### PR DESCRIPTION
**Pull request by kaiserls**
# Description

Fixes #38 by using normalized and uncorrelated data internally. Adds a DataTransformation class to handle the transformations. Uses jax to fix #52. Implements the PCA to fix #54 .

## Type of change

Please tick at least one of the options. Delete the other options.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?

- [x] Added a test for the Identity transformation, the normalization transformation and a first run tests for the pca transformation.

## Checklist For Contributor

- [x] My code follows the style guidelines of this project (I installed pre-commit)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I wrote my changes in the changelog in the `[unreleased]` section
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes

## Checklist For Maintainers

- [x] Continuous Integration (CI) is successfully running
- [x] Do we want to release/tag a new version? [✔️] v0.3.0
  - [x] If yes, add a release to the changelog and set the new version in pyproject.toml. After the merge, tag the release!

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,16 @@ All notable changes to this project will be documented in this file.
 
 ## [Unreleased]
 
+## [0.3.0] - 2023-04-27
+
+### Added
+
+- Added data normalization to improve numerics and performance in general
```
Comment (castellinilinguini) [inline CHANGELOG.md:27]: PCA should probably be added to the changelog

File: eulerpi/core/data_transformation.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    def transform(self, data: jnp.ndarray) -> jnp.ndarray:
+        """Transform the given data using the PCA.
+
+        Args:
+            data (jnp.ndarray): The data to be transformed.
+
+        Returns:
+            jnp.ndarray: The data projected onto and expressed in the basis of the principal components.
+        """
+        return self.pca.transform(data.reshape(-1, data.shape[-1])).reshape(
```
Comment (castellinilinguini) [inline eulerpi/core/data_transformation.py:212]: We might want to change the orientation of the data internally in the future.

**Review by castellinilinguini — CHANGES_REQUESTED** [review]
Please update the changelog, otherwise awesome pull request!

File: eulerpi/core/data_transformation.py (hunk 2 above)
Comment (kaiserls) [reply eulerpi/core/data_transformation.py:212]: Yep, also using basic numpy functions like matmul instead of inner would be amazing.

**Review by castellinilinguini — APPROVED** [review]
LGTM!
